pipelines/validate_vset.py

```
@@ -0,0 +1,894 @@
+    rows = sorted(result.rows, key=lambda item: item["id"])
+    report = {
+        "tests": rows,
+        "failures": len(result.failures),
+        "errors": len(result.errors),
+        "ok": result.wasSuccessful(),
+    }
+    report["result_hash"] = _sha256_text(_canonical_json(report["tests"]))
+    return report
+
+
+def apply_patch(work: Path, patch: Any) -> None:
```

Ruff (0.16.2)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `patch`

(ANN401)